service/src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -1,55 +1,72 @@
 import bio.terra.stairway.Flight;
 import bio.terra.stairway.FlightMap;
+import bio.terra.workspace.common.utils.MdcHook;
 import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.exception.InvalidJobIdException;
 import bio.terra.workspace.service.job.exception.InvalidJobParameterException;
 import io.opencensus.contrib.spring.aop.Traced;
+import org.apache.commons.lang3.StringUtils;
 
 public class JobBuilder {
-
-  private final JobService jobServiceRef;
-  private final Class<? extends Flight> flightClass;
+  private final JobService jobService;
+  private final StairwayComponent stairwayComponent;
+  private final MdcHook mdcHook;
   private final FlightMap jobParameterMap;
-  private final String jobId;
-
-  // constructor only takes required parameters
-  public JobBuilder(
-      String description,
-      String jobId,
-      Class<? extends Flight> flightClass,
-      Object request,
-      AuthenticatedUserRequest userRequest,
-      JobService jobServiceRef) {
-    this.jobServiceRef = jobServiceRef;
+  private Class<? extends Flight> flightClass;
+  private String jobId;
+  private String description;
+  private Object request;
+  private AuthenticatedUserRequest userRequest;
+
+  public JobBuilder(JobService jobService, StairwayComponent stairwayComponent, MdcHook mdcHook) {
+    this.jobService = jobService;
+    this.stairwayComponent = stairwayComponent;
+    this.mdcHook = mdcHook;
+    this.jobParameterMap = new FlightMap();
+  }
+
+  public JobBuilder flightClass(Class<? extends Flight> flightClass) {
     this.flightClass = flightClass;
+    return this;
+  }
+
+  public JobBuilder jobId(String jobId) {
+    if (jobId != null) {
+      // If clients provide a non-null job ID, it cannot be whitespace-only
+      if (StringUtils.isWhitespace(jobId)) {
+        throw new InvalidJobIdException("jobId cannot be whitespace-only.");
+      }
+    }
     this.jobId = jobId;
+    return this;
+  }
 
-    // initialize with required parameters
-    this.jobParameterMap = new FlightMap();
-    jobParameterMap.put(JobMapKeys.DESCRIPTION.getKeyName(), description);
-    jobParameterMap.put(JobMapKeys.REQUEST.getKeyName(), request);
-    jobParameterMap.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
-    jobParameterMap.put(JobMapKeys.SUBJECT_ID.getKeyName(), userRequest.getSubjectId());
+  public JobBuilder description(String description) {
+    this.description = description;
+    return this;
+  }
+
+  public JobBuilder request(Object request) {
+    this.request = request;
+    return this;
+  }
+
+  public JobBuilder userRequest(AuthenticatedUserRequest userRequest) {
+    this.userRequest = userRequest;
+    return this;
   }
 
-  // use addParameter method for optional parameter
-  // returns the JobBuilder object to allow method chaining
   public JobBuilder addParameter(String keyName, Object val) {
     if (keyName == null) {
```

Comment:
do we need to check if val is nullable?

Also StringUtils.isBlank(keyName) instead?